pkg/cvo/metrics.go

```diff
@@ -127,15 +132,69 @@ type asyncResult struct {
 	error error
 }
 
-func createHttpServer() *http.Server {
+func createHttpServer(ctx context.Context, client *authenticationclientsetv1.AuthenticationV1Client) *http.Server {
+	auth := authHandler{downstream: promhttp.Handler(), ctx: ctx, client: client.TokenReviews()}
 	handler := http.NewServeMux()
-	handler.Handle("/metrics", promhttp.Handler())
+	handler.Handle("/metrics", &auth)
 	server := &http.Server{
 		Handler: handler,
 	}
 	return server
 }
 
+type tokenReviewInterface interface {
+	Create(ctx context.Context, tokenReview *authenticationv1.TokenReview, opts metav1.CreateOptions) (*authenticationv1.TokenReview, error)
+}
+
+type authHandler struct {
+	downstream http.Handler
+	ctx        context.Context
+	client     tokenReviewInterface
+}
+
+func (a *authHandler) authorize(token string) (bool, error) {
+	tr := &authenticationv1.TokenReview{
+		Spec: authenticationv1.TokenReviewSpec{
+			Token: token,
+		},
+	}
+	result, err := a.client.Create(a.ctx, tr, metav1.CreateOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to check token: %w", err)
+	}
+	if !result.Status.Authenticated {
+		klog.V(4).Info("The token cannot be authenticated.")
+	} else if user := result.Status.User.Username; user != "system:serviceaccount:openshift-monitoring:prometheus-k8s" {
+		klog.V(4).Infof("Access the metrics from the unexpected user %s is denied.", user)
+	}
+	return result.Status.Authenticated && result.Status.User.Username == "system:serviceaccount:openshift-monitoring:prometheus-k8s", nil
+}
+
+func (a *authHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	authHeader := r.Header.Get("Authorization")
+	if authHeader == "" {
+		http.Error(w, "failed to get the Authorization header", http.StatusUnauthorized)
+		return
+	}
+	token := strings.TrimPrefix(authHeader, "Bearer ")
+	if token == authHeader {
```

Comment:
this technically lets the empty token string through to `auth` (where it correctly fails) but that's just me being pedantic

Reply:
Done.